### PR TITLE
[Jenkinsfile] Update CMS Tugboat config and fix broken link checker failing when it shouldn't

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR updates our FE<->CMS environments to always use the CMS's Tugboat instance, unless the build is a content-only build or if it is the production build on master.

https://github.com/department-of-veterans-affairs/va.gov-cms/issues/5207

## Testing done
- To fully test the Tugboat config, will need to validate CMS prod URL is visible in the output of the master branch's prod build.
- To test the updates to the broken link checker, I created a broken link in a layout file then observed that the Slack message was triggered but the build reported no errors

## Screenshots
N/A

This screenshot shows the broken link checker triggering a Slack message but the build still passing

image

## Acceptance criteria
- [ ] CMS Prod instance no longer tops out

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by acrollet — COMMENTED** [review]
This looks like what we want, I'll defer to @indytechcook for an approval though!

**Review by indytechcook — APPROVED** [review]

**Review by theodur — APPROVED** [review]

**Pull request merged**
